allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java

```diff
@@ -75,7 +77,12 @@ public Allure2ResultsReader(AttachmentsStorage storage) {
                     TestCaseResult dest = new TestCaseResult();
                     dest.setUid(generateUid());
 
-                    dest.setId(result.getId());
+                    String id = Optional.ofNullable(result.getId()).orElseGet(() -> {
```

Comment:
This is where test case result with null id is being parsed. We discussed that we wand to generate an id if it has not been provided, thus making this test an orphan, aren't we?